### PR TITLE
TOMEE-4117 OpenAPI model not being generated from annotations.

### PR DESCRIPTION
**Pull request by jeanouii**

### CODE REVIEW

File: tomee/tomee-microprofile/mp-common/src/main/java/org/apache/tomee/microprofile/openapi/MicroProfileOpenApiRegistration.java [hunk 1]
```diff
@@ -16,76 +16,202 @@
...
+    }
+
+    /**
+     * Provides the Jandex index.
+     */
+    private static Index index(final Set<Class<?>> classes, final ServletContext servletContext, final OpenApiConfig config) {
+        FilteredIndexView filteredIndexView = new FilteredIndexView(null, config);
+        Indexer indexer = new Indexer();
+
+        // todo load all classes and build up an index with the content
```
Comment (rzo1) [inline tomee/tomee-microprofile/mp-common/src/main/java/org/apache/tomee/microprofile/openapi/MicroProfileOpenApiRegistration.java:136]: still an todo?

**Review by rzo1 — APPROVED** [review]

**Comment by rzo1** [comment]
Looks like something is weird with this branch:

```bash
Error:  Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.6.2:compile (default-compile) on project tomee-security-itests: Compilation failure: Compilation failure: 
Error:  /home/runner/work/tomee/tomee/itests/tomee-security-itests/src/main/java/org/apache/tomee/security/itest/AnotherAuthMechanism.java:[31,38] package org.apache.tomee.security.http does not exist
Error:  /home/runner/work/tomee/tomee/itests/tomee-security-itests/src/main/java/org/apache/tomee/security/itest/AnotherAuthMechanism.java:[38,75] cannot find symbol
Error:    symbol: class LoginToContinueMechanism
Error:  /home/runner/work/tomee/tomee/itests/tomee-security-itests/src/main/java/org/apache/tomee/security/itest/AnotherAuthMechanism.java:[68,5] method does not override or implement a method from a supertype
Error:  -> [Help 1]
```

**Comment by jeanouii** [comment]
Thanks for the review Richard. I'll have a look

**Comment by jeanouii** [comment]
I've got some regressions to look after. So don't merge it quite yet

**Pull request merged**